test/features/DebugSession.test.ts

```diff
@@ -0,0 +1,491 @@
+
+            const debugAdapterDescriptor = await debugSessionFeature.createDebugAdapterDescriptor(debugSession, undefined);
+
+            // Confirm debugAdapterDescriptor is of type debugadapternamedpipeserver
+            assert.ok(debugAdapterDescriptor instanceof DebugAdapterNamedPipeServer);
+            assert.equal(debugAdapterDescriptor.path, "testPipeName");
+        });
+    });
+});
+
+describe("DebugSessionFeature E2E", function slowTests() {
+    this.slow(20000); // Will warn if test takes longer than 10s and show red if longer than 20s
+
+    if (process.platform == "darwin") {
+        this.timeout(60000); // The MacOS test runner is sloooow in Azure Devops
+    }
+    before(async () => {
+        // Registers and warms up the debug adapter and the PowerShell Extension Terminal
+        await ensureEditorServicesIsConnected();
+    });
+
+    it("Starts and stops a debugging session", async () => {
+        // Inspect the debug session via the started events to ensure it is correct
+        let startDebugSession: DebugSession;
+        let stopDebugSession: DebugSession;
+        const interactiveSessionConfig = defaultDebugConfigurations[DebugConfig.InteractiveSession];
+        // Asserts dont seem to fire in this event or the event doesnt resolve in the test code flow, so we need to "extract" the values for later use by the asserts
+
+        const startDebugEvent = debug.onDidStartDebugSession((newDebugSession) => {
+            startDebugEvent.dispose();
+            startDebugSession = newDebugSession;
+            const stopDebugEvent = debug.onDidTerminateDebugSession((terminatedDebugSession) => {
+                stopDebugEvent.dispose();
+                stopDebugSession = terminatedDebugSession;
+            });
+        });
+
+        const debugSessionStarted = await debug.startDebugging(undefined, interactiveSessionConfig);
+        assert.ok(debugSessionStarted, "Debug session should start");
+        assert.equal(startDebugSession!.name, interactiveSessionConfig.name, "Debug session name should match when started");
+        // debugSession var should be populated from the event before startDebugging completes
+        await debug.stopDebugging(startDebugSession!);
+
+        assert.equal(stopDebugSession!.name, interactiveSessionConfig.name, "Debug session name should match when stopped");
+        assert.equal(stopDebugSession!.configuration.internalConsoleOptions, "neverOpen", "Debug session should always have neverOpen internalConsoleOptions");
+        assert.ok(stopDebugSession!, "Debug session should stop");
+    });
+
+    describe("Binary Modules", () => {
+        before(async () => {
+            await InstallCSharpExtension();
+            BuildBinaryModuleMock();
+            await ensureEditorServicesIsConnected();
+        });
+        afterEach(async () => {
+            // await debug.stopDebugging(undefined);
+            // await commands.executeCommand("workbench.action.closeAllEditors");
```

Comment:
Did you mean to leave these disabled?

Reply:
No, MacOS was driving me crazy, I think both of these are idempotent so they should be OK to run even if no debuggers are running. Let me fix everything else and I'll come back to this.

Reply:
719275b697243d82a59193c30ac4bb5948aaf3bc I removed the closeAllEditors because it's currently not relevant to tests anymore since I switched to a different way of testing the breakpoint being hit.